JuliaInterface/gap/JuliaInterface.gi

```diff
@@ -57,11 +57,74 @@ InstallMethod( CallFuncList,
     end );
 
 
-##
-##  We support '<juliaobj>[<i>]' and '<juliaobj>[<i>, <j>]' in general.
-##  This is useful for example if <juliaobj> is a tuple.
-##
-BindJuliaFunc( "getindex", "Base" );
+BindGlobal( "_JULIA_MODULE_TYPE", JuliaEvalString( "Module" ) );
+BindGlobal( "_JULIA_FUNCTION_TYPE", JuliaEvalString( "Function" ) );
+BindGlobal( "_JULIA_ISA", JuliaFunction( "isa" ) );
+
+BindGlobal( "_NewJuliaModule",
+  function( name, julia_pointer )
+    local module;
+
+    module := rec( storage := rec( ),
+                   julia_pointer := julia_pointer );
+
+    Objectify( TheTypeOfJuliaModules, module );
+
+    SetName( module, name );
+
+    return module;
+
+end );
+
+InstallMethod( \.,
+              [ IsJuliaModule, IsPosInt ],
+  function( module, rnum )
+    local rnam, global_variable;
+
+    if IsBound\.( module!.storage, rnum ) then
+        return \.(module!.storage, rnum );
+    fi;
+
+    rnam := NameRNam( rnum );
+
+    global_variable := _JuliaGetGlobalVariableByModule( rnam, module!.julia_pointer );
+
+    if ConvertedFromJulia( _JULIA_ISA( global_variable, _JULIA_FUNCTION_TYPE ) ) then
+        global_variable := _JuliaFunction( global_variable );
+    elif ConvertedFromJulia( _JULIA_ISA( global_variable, _JULIA_MODULE_TYPE ) ) then
+        global_variable := _NewJuliaModule( rnam, global_variable );
+    fi;
+
+    \.\:\=( module!.storage, rnum, global_variable );
+    return global_variable;
+
+end );
+
+InstallMethod( \.\:\=,
+               [ IsJuliaModule, IsPosInt, IsObject ],
+  function( module, rnum, obj )
+    \.\:\=( module!.storage, rnum, obj );
+end );
+
+
+InstallMethod( IsBound\.,
+               [ IsJuliaModule, IsPosInt ],
+    {module,rnum} -> IsBound\.( module!.storage, rnum  ) );
```

Comment:
Hmm, that's not the semantic I wold expect. I'd expect this to look more like this:
```gap
function(module, rnum)
  if IsBound\.( module!.storage, rnum  ) then return true; fi;
  rnam := NameRNam( rnum );
  # test if Julia knows a variable with that name
  return fail <> _JuliaGetGlobalVariableByModule( rnam, module!.julia_pointer );
end;
```
Well, this assumes that `_JuliaGetGlobalVariableByModule` returns fail if there is no such variable, which may be wrong (I did not check what it does), but I hope the general idea is clear.

Reply:
Yep, that makes more sense, as the binding happens automatically. Did not think about this.